firedrake/fml/form_manipulation_language.py

```diff
@@ -55,36 +57,40 @@ def get(self, label):
         """
         return self.labels.get(label.label)
 
-    def has_label(self, *labels, return_tuple=False):
+    def has_label(
+        self,
+        *labels: Union[Sequence["Label"], "Label"],
```

Comment:
This says each entry in `labels` is either a single label, or a sequence of labels. You want:
```
*labels: "Label"
```
I suspect.